### PR TITLE
Add connector.Builder

### PR DESCRIPTION
**Pull request by djaglowski**

### CODE REVIEW

**Comment by djaglowski** [comment]
@bogdandrutu, this PR is ready for review.

File: connector/connector.go [hunk 1]
```diff
@@ -453,3 +458,173 @@ func NewFactory(cfgType component.Type, createDefaultConfig component.CreateDefa
...
+// NewBuilder creates a new connector.Builder to help with creating components form a set of configs and factories.
+func NewBuilder(cfgs map[component.ID]component.Config, factories map[component.Type]Factory) *Builder {
+	return &Builder{cfgs: cfgs, factories: factories}
+}
+
+// CreateTracesToTraces creates a Traces connector based on the settings and config.
+func (b *Builder) CreateTracesToTraces(ctx context.Context, set CreateSettings, next consumer.Traces) (Traces, error) {
+	cfg, existsCfg := b.cfgs[set.ID]
+	if !existsCfg {
+		return nil, fmt.Errorf("connector %q is not configured", set.ID)
```
Comment (bogdandrutu) [inline connector/connector.go:477]: Would be good to have some tests for all these if cases.

**Review by bogdandrutu — APPROVED** [review]

File: connector/connector.go (hunk 1 above)
Comment (djaglowski) [reply connector/connector.go:490]: Good call. I've added tests equivalent to those in #6862

**Comment by djaglowski** [comment]
This should be ready to merge.

**Comment by bogdandrutu** [comment]
Please rebase

**Comment by djaglowski** [comment]
Rebased

**Pull request merged**